Media/MediaPlayer.cpp

```diff
@@ -1303,14 +1306,40 @@ int AudioBufferDataSource::read_packet(void *opaque, uint8_t *buf,
 int AudioBufferDataSource::ReadPacket(uint8_t *buf, int buf_size) {
     int size = buf_end - buf_pos;
     if (size <= 0) {
-        return 0;
+        return AVERROR_EOF;
     }
     size = std::min(buf_size, size);
     memcpy(buf, buf_pos, size);
     buf_pos += size;
     return size;
 }
 
+int64_t AudioBufferDataSource::seek(void *opaque, int64_t offset, int whence) {
+    AudioBufferDataSource *pThis = (AudioBufferDataSource *)opaque;
+    return pThis->Seek(opaque, offset, whence);
+}
+
+int64_t AudioBufferDataSource::Seek(void *opaque, int64_t offset, int whence) {
+    if ((whence & AVSEEK_SIZE) == AVSEEK_SIZE) {
+        return buffer->GetSize();
+    }
+    int force = whence & AVSEEK_FORCE;
+    whence &= ~AVSEEK_FORCE;
+    whence &= ~AVSEEK_SIZE;
+    uint8_t *buf_start = (uint8_t*)buffer->GetData();
+    if (whence == SEEK_SET) {
+        buf_pos = std::clamp(buf_start + offset, buf_start, buf_end);
+        return buf_pos - buf_start;
+    } else if (whence == SEEK_CUR) {
+        buf_pos = std::clamp(buf_pos + offset, buf_start, buf_end);
+        return buf_pos - buf_start;
+    } else if (whence == SEEK_END) {
+        buf_pos = std::clamp(buf_end + offset, buf_start, buf_end);
+        return buf_pos - buf_start;
+    }
+    return AVERROR(EIO);
```

Comment:
Cosmetic: I'd suggest using AVERROR_IO here.

Reply:
What is AVERROR_IO ?

Reply:
There is a #define in ffmpeg code: 
```
#define AVERROR_IO AVERROR(EIO)
```

So I'm assuming this is what we're supposed to use in user code. Same as with AVERROR_EOF above.

Reply:
Hm...
Have no this define in 4.2.2

Reply:
Eh, indeed. My comment doesn't apply then.